### PR TITLE
adding contact page + button back

### PR DESCRIPTION
**Pull request by sorrento974**

### CODE REVIEW

File: client/src/styles/Contact.css [hunk 1]
```diff
@@ -0,0 +1,135 @@
+.logo-pixhunt {
+  width: 7rem;
+  display: flex;
+  margin: 0.5rem 0 0 0.3rem;
+}
+.contact-form-h2 {
+  text-align: center;
+  color: var(--third-color);
+  margin: 0rem 0 1.25rem 0;
+  font-family: "Atkinson Hyperlegible", sans-serif;
```
Comment (Delgorithm) [inline client/src/styles/Contact.css]: You have to change the variable with the ones in the root

File: client/src/styles/Contact.css [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+.form-group {
+  margin-bottom: 15px;
+}
+
+.form-group label {
+  display: block;
+  margin-bottom: 5px;
+  font-weight: bold;
```
Comment (Delgorithm) [inline client/src/styles/Contact.css:41]: You'll have to change the "px" to "rem"

File: client/src/styles/Contact.css [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  font-family: "Atkinson Hyperlegible", sans-serif;
+}
+
+.form-control {
+  width: 100%;
+  padding: 10px;
+  border: 1px solid #ccc;
```
Comment (Delgorithm) [inline client/src/styles/Contact.css]: You'll have to change the "px" to "rem"

File: client/src/styles/Contact.css [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  border-radius: 4px;
+  resize: none;
+}
+
+.form-control::placeholder {
+  font-family: "Atkinson Hyperlegible", sans-serif;
+  color: #bbb;
```
Comment (Delgorithm) [inline client/src/styles/Contact.css]: You have to change the variable with the ones in the root

File: client/src/styles/Contact.css [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+.error-message {
+  color: red;
+  font-size: 0.875em;
+  margin-top: 5px;
+}
```
Comment (Delgorithm) [inline client/src/styles/Contact.css:58]: You'll have to change the "px" to "rem"

File: client/src/styles/Contact.css [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+  margin: auto;
+  padding: 0.7rem;
+  font-size: 1.2rem;
+  background-color: var(--secondary-color);
+  color: var(--primary-color);
+  border: none;
+  border-radius: 4px;
+  cursor: pointer;
+  transition: background-color 0.3s ease; /* a modifier */
+  font-family: "Atkinson Hyperlegible", sans-serif;
```
Comment (Delgorithm) [inline client/src/styles/Contact.css]: You have to delete the comment

File: client/src/styles/Contact.css [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+.submit-btn:disabled {
+  background-color: #c0c0c0;
+  cursor: not-allowed;
+}
+
+.submit-btn:hover:not(:disabled) {
+  /* background-color: #0056b3; */
+  scale: 0.95;
```
Comment (Delgorithm) [inline client/src/styles/Contact.css:75]: You have to delete the comment

File: client/src/styles/Contact.css [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+.alert {
+  margin-top: 20px;
+}
```
Comment (Delgorithm) [inline client/src/styles/Contact.css:80]: You'll have to change the "px" to "rem"

**Review by Delgorithm — CHANGES_REQUESTED** [review]

File: client/src/styles/Contact.css [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+  margin-bottom: 1.25rem;
+}
+
+.form-fields {
+  list-style: none;
+  padding: 0;
+}
+
+.form-group {
+  margin-bottom: 15px;
```
Comment (Alice-Berthelot) [inline client/src/styles/Contact.css]: there is a px that should be transformed into rem (line 34)

File: client/src/styles/Contact.css [hunk 10]
```diff
@@ -36,28 +36,28 @@
...
+  font-family: var(--secondary-font);
 }
 
 .form-control {
   width: 100%;
-  padding: 10px;
-  border: 1px solid #ccc;
-  border-radius: 4px;
+  padding: 0.6rem;
+  border: 1px solid #bbb;
```
Comment (Alice-Berthelot) [inline client/src/styles/Contact.css]: you should choose colors that are in our variables, not #bbb

File: client/src/styles/Contact.css [hunk 11]
```diff
@@ -36,28 +36,28 @@
...
+  border-radius: 0.15rem;
   resize: none;
 }
 
 .form-control::placeholder {
-  font-family: "Atkinson Hyperlegible", sans-serif;
+  font-family: var(--secondary-font);
   color: #bbb;
```
Comment (Alice-Berthelot) [inline client/src/styles/Contact.css]: you should choose colors that are in our variables, not #bbb

File: client/src/styles/Contact.css [hunk 12]
```diff
@@ -36,28 +36,28 @@
...
 }
 
 .error-message {
   color: red;
```
Comment (Alice-Berthelot) [inline client/src/styles/Contact.css:55]: you should choose colors that are in our variables, not #bbb

File: client/src/styles/Contact.css [hunk 13]
```diff
@@ -70,8 +70,8 @@
   border: none;
   border-radius: 4px;
```
Comment (Alice-Berthelot) [inline client/src/styles/Contact.css]: the px should be transformed into rem

File: client/src/styles/Contact.css [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+.contact-form-list {
+  list-style: none;
+  margin: 1rem;
+  max-width: 25rem;
+  background-color: var(--primary-color);
+  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
+  padding: 1.25rem;
+  border-radius: 10px;
```
Comment (Alice-Berthelot) [inline client/src/styles/Contact.css]: px should be transformed into rem

File: client/src/styles/Contact.css [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+
+.submit-btn {
+  display: block;
+  margin: auto;
+  padding: 0.7rem;
+  font-size: 1.2rem;
+  background-color: var(--secondary-color);
+  color: var(--primary-color);
+  border: none;
+  border-radius: 4px;
```
Comment (Alice-Berthelot) [inline client/src/styles/Contact.css]: px should be transformed into rem

File: client/src/styles/Contact.css [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+  cursor: pointer;
+  transition: background-color 0.3s ease;
+  font-family: var(--secondary-font);
+}
+
+.submit-btn:disabled {
+  background-color: #c0c0c0;
```
Comment (Alice-Berthelot) [inline client/src/styles/Contact.css]: you should pick colours from our variables only

**Review by Alice-Berthelot — CHANGES_REQUESTED** [review]

**Review by Alice-Berthelot — APPROVED** [review]

File: client/src/pages/Contact.jsx [hunk 17]
```diff
@@ -0,0 +1,187 @@
...
+      await emailjs.send(
+        process.env.REACT_APP_SERVICE_ID,
+        process.env.REACT_APP_TEMPLATE_ID,
+        templateParams,
+        process.env.REACT_APP_USER_ID
+      );
+
+      toggleAlert("La soumission du formulaire a fonctionné", "success");
+    } catch (e) {
+      console.error(e);
```
Comment (KevinDavoust) [inline client/src/pages/Contact.jsx]: rien dans la console

File: client/src/pages/Contact.jsx [hunk 18]
```diff
@@ -0,0 +1,187 @@
...
+            />
+            {errors.message && (
+              <p className="error-message">{errors.message.message}</p>
+            )}
+          </li>
+        </form>
+
+        {alertInfo.display && (
+          <li className="contact-form-item">
+            <div
```
Comment (KevinDavoust) [inline client/src/pages/Contact.jsx]: pas ça zinédine... pas après tout ce que tu as fait.......

File: client/src/pages/Contact.jsx [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+        aria-labelledby="submit-btn"
+        className="submit-btn external-submit-btn"
+        disabled={disabled}
+        onClick={handleExternalSubmit}
+      >
+        Envoyer
+      </button>
+      <img
+        className="contact-form-image"
+        src="./src/assets/images/5ba7c8d79a1de659493719.jpg"
```
Comment (KevinDavoust) [inline client/src/pages/Contact.jsx]: il faudrait importer l'image comme une variable en haut du fichier plutôt que mettre le chemin en dur directement dans src

**Review by KevinDavoust — CHANGES_REQUESTED** [review]

File: client/src/components/BtnGoBack.jsx [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+import { BsArrowLeft } from "react-icons/bs";
+import { useNavigate } from "react-router-dom";
+
+function BtnGoBack() {
+  const navigate = useNavigate();
+  const handleButtonClick = () => {
+    navigate("/");
+  };
+  return (
+    <button
```
Comment (KevinDavoust) [inline client/src/components/BtnGoBack.jsx:11]: vu que ce bouton exécute une redirection, une balise Link serait plus appropriée, et éviterait cette méthode handleClick

File: client/src/pages/Contact.jsx [hunk 21]
```diff
@@ -0,0 +1,184 @@
...
+              data-bs-dismiss="alert"
+              aria-label="Close"
+              onClick={() =>
+                setAlertInfo({ display: false, message: "", type: "" })
+              }
+            />
+          </li>
+        )}
+        <button
+          type="button"
```
Comment (KevinDavoust) [inline client/src/pages/Contact.jsx]: le bouton doit être de type submit pour soumettre le formulaire, le type button est réservé pour de simples interactions utilisateur

**Review by KevinDavoust — CHANGES_REQUESTED** [review]

File: client/src/pages/Contact.jsx (hunk 21 above)
Comment (Alice-Berthelot) [reply client/src/pages/Contact.jsx]: done

File: client/src/components/BtnGoBack.jsx
Comment (Alice-Berthelot) [inline client/src/components/BtnGoBack.jsx:1]: this button should link to the previous page, not home

**Review by Alice-Berthelot — CHANGES_REQUESTED** [review]

**Pull request merged**